app/models/certificate.rb

```diff
@@ -15,6 +15,34 @@ class Certificate < Sequel::Model
       .order(Sequel.desc(:certificate_description_periods__validity_start_date))
   end
 
+  dataset_module do
+    def q_search(filter_ops)
+      scope = actual
+
+      if filter_ops[:q].present?
+        q_rule = "#{filter_ops[:q]}%"
+
+        scope = scope.join_table(:inner,
+          :certificate_descriptions,
+          certificate_type_code: :certificate_type_code,
+          certificate_code: :certificate_code
+        ).where("
+          certificates.certificate_code ilike ? OR
```

Comment:
could be written in sequel SQL::Expression's

http://sequel.jeremyevans.net/rdoc/files/doc/querying_rdoc.html#label-SQL-3A-3AExpression

With string concat https://sequel.jeremyevans.net/rdoc/files/doc/dataset_filtering_rdoc.html#label-String+concatenation

Reply:
@matthewford agree will rewrite